airbyte-scheduler/src/main/java/io/airbyte/scheduler/ScheduleJobPredicate.java

```diff
@@ -57,17 +57,11 @@ private boolean shouldSchedule(Optional<Job> previousJobOptional, boolean timeFo
     }
 
     final Job previousJob = previousJobOptional.get();
-    switch (previousJob.getStatus()) {
-      case CANCELLED:
-      case SUCCEEDED:
-        return timeForJobNewJob;
-      case FAILED:
-      case PENDING:
-      case RUNNING:
-        return false;
-    }
-
-    return false;
+    return switch (previousJob.getStatus()) {
+      case CANCELLED, SUCCEEDED, FAILED -> timeForJobNewJob;
+      case INCOMPLETE, PENDING, RUNNING -> false;
```

Comment:
Why is incomplete not rescheduling a job?

Does it seem like it would result in this issue if I am not wrong?

https://github.com/airbytehq/airbyte/issues/1425